zebra-chain/src/block/tests/generate.rs

```diff
@@ -125,8 +226,112 @@ fn single_transaction_block(oversized: bool) -> Block {
+        header: block_header,
         transactions,
+    };
+
+    let serialized_len = block.zcash_serialize_to_vec().unwrap().len();
+    assert_eq!(
+        oversized,
+        serialized_len > MAX_BLOCK_BYTES.try_into().unwrap(),
+        "block is over-sized if requested:\n\
+         oversized: {},\n\
+         serialized_len: {},\n\
+         MAX_BLOCK_BYTES: {},",
+        oversized,
+        serialized_len,
+        MAX_BLOCK_BYTES,
+    );
+    assert!(
+        serialized_len > MIN_LARGE_BLOCK_BYTES.try_into().unwrap(),
+        "block is large\n\
+         oversized: {},\n\
+         serialized_len: {},\n\
+         MIN_LARGE_BLOCK_BYTES: {},",
+        oversized,
+        serialized_len,
+        MIN_LARGE_BLOCK_BYTES,
+    );
+
+    block
+}
+
+/// Implementation of block generation with one transaction and multiple transparent outputs
+fn single_transaction_block_many_outputs(oversized: bool) -> Block {
+    // Dummy input and output
+    let (input, input_bytes) = input();
+    let (output, output_bytes) = output();
+
+    // A block header
+    let (block_header, block_header_bytes) = block_header();
+
+    // A LockTime
+    let (lock_time, lock_time_bytes) = lock_time();
+
+    // Calculate the number of outputs we need,
+    // subtracting the bytes used to serialize the expected output count,
+    // transaction count, and input count.
+    let mut max_outputs_in_tx = (usize::try_from(MAX_BLOCK_BYTES).unwrap()
+        - block_header_bytes.len()
+        - 1
+        - TX_V1_HEADER_BYTES
+        - lock_time_bytes.len()
+        - 1
+        - input_bytes.len()
+        - MAX_COMPACT_SIZE_BYTES)
+        / output_bytes.len();
+
+    if oversized {
+        max_outputs_in_tx += 1;
     }
+
+    // 1 single input
+    let inputs = vec![input];
+
+    // Create outputs to be just below the limit
+    let outputs = std::iter::repeat(output)
+        .take(max_outputs_in_tx)
+        .collect::<Vec<_>>();
+
+    // Create a big transaction
+    let big_transaction = Transaction::V1 {
+        inputs,
+        outputs,
+        lock_time,
+    };
+
+    // Put the big transaction into a block
+    let transactions = vec![Arc::new(big_transaction)];
+
+    let block = Block {
+        header: block_header,
+        transactions,
+    };
+
+    let serialized_len = block.zcash_serialize_to_vec().unwrap().len();
+    assert_eq!(
+        oversized,
+        serialized_len > MAX_BLOCK_BYTES.try_into().unwrap(),
+        "block is over-sized if requested:\n\
+         oversized: {},\n\
+         serialized_len: {},\n\
+         MAX_BLOCK_BYTES: {},",
+        oversized,
+        serialized_len,
+        MAX_BLOCK_BYTES,
+    );
+    assert!(
+        serialized_len > MIN_LARGE_BLOCK_BYTES.try_into().unwrap(),
+        "block is large\n\
+         oversized: {},\n\
+         serialized_len: {},\n\
+         MIN_LARGE_BLOCK_BYTES: {},",
+        oversized,
+        serialized_len,
+        MIN_LARGE_BLOCK_BYTES,
```

Comment:
Suggestion: small tweak for conciseness.

```suggestion
         oversized: {oversized},\n\
         serialized_len: {serialized_len},\n\
         MIN_LARGE_BLOCK_BYTES: {MIN_LARGE_BLOCK_BYTES},",
```